clang/unittests/Format/TokenAnnotatorTest.cpp

```diff
@@ -720,8 +720,9 @@ TEST_F(TokenAnnotatorTest, UnderstandsNonTemplateAngleBrackets) {
 
   Tokens = annotate("return A < B != A > B;");
   ASSERT_EQ(Tokens.size(), 10u) << Tokens;
-  EXPECT_TOKEN(Tokens[2], tok::less, TT_BinaryOperator);
-  EXPECT_TOKEN(Tokens[6], tok::greater, TT_BinaryOperator);
+  // FIXME:
```

Comment:
Just a thought, could we check the token after `greater`? If it is an identifier it can never be a template.

Reply:
It can be a template closer e.g. in declarations with implicit instantiated template types? IMO we should rework `parseAngle()` instead of adding more heuristics for special cases such as `A<B != C> D` that must be an expression if preceded by `return`.